### PR TITLE
[dbg-macro] conan v2 migration

### PR DESCRIPTION
**Pull request by Doome161**
Specify library name and version:  **dbg-macro/0.4.0**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/dbg-macro/all/test_v1_package/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+cmake_minimum_required(VERSION 3.1)
+project(test_package LANGUAGE CXX)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup()
+
+add_executable(${PROJECT_NAME} ../test_package/test_package.cpp)
+target_link_libraries(${PROJECT_NAME} ${CONAN_LIBS})
+set_property(TARGET ${PROJECT_NAME} PROPERTY CXX_STANDARD 11)
```
Comment (uilianries) [inline recipes/dbg-macro/all/test_v1_package/CMakeLists.txt:9]:
```suggestion
conan_basic_setup(TARGETS)

add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/../test_package/
                 ${CMAKE_CURRENT_BINARY_DIR}/test_package/)
```
Re-use the Cmake file from test_package folder.

File: recipes/dbg-macro/all/test_v1_package/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
+from conans import ConanFile, CMake, tools
+from conan.tools.build import cross_building
+import os
+
+
+# legacy validation with Conan 1.x
```
Comment (uilianries) [inline recipes/dbg-macro/all/test_v1_package/conanfile.py:6]:
```suggestion
```

File: recipes/dbg-macro/all/test_package/conanfile.py [hunk 3]
```diff
@@ -1,18 +1,27 @@
+from conan import ConanFile
+from conan.tools.build import can_run
+from conan.tools.cmake import cmake_layout, CMake
 import os
-from conans import ConanFile, CMake, tools
 
 
+# It will become the standard on Conan 2.x
```
Comment (uilianries) [inline recipes/dbg-macro/all/test_package/conanfile.py:7]:
```suggestion
```

**Review by uilianries — COMMENTED** [review]
Thank you for your contribution

File: recipes/dbg-macro/all/test_v1_package/CMakeLists.txt (hunk 1 above)
Comment (Doome161) [reply recipes/dbg-macro/all/test_v1_package/CMakeLists.txt]: Thanks for the suggestion! That should also be updated in the templates

Comment (uilianries) [reply recipes/dbg-macro/all/test_v1_package/CMakeLists.txt]: Which template is outdated? We have updated all them.

Comment (Doome161) [reply recipes/dbg-macro/all/test_v1_package/CMakeLists.txt]: Nice! it was just my local copy that wasn't up to date yet

**Pull request closed without merging**